File: new-algorithm/src/javaalgo/string/boj11656.java
package javaalgo.string;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.TreeSet;

public class boj11656 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        String str = br.readLine();

        TreeSet<String> set = new TreeSet<>();
        for (int i = 0; i < str.length(); i++) {
            set.add(str.substring(i));
        }

        for (String s : set) {
            sb.append(s).append('\n');
        }

        System.out.println(sb);
    }
}

File: new-algorithm/src/kotlinalgo/string/boj11656.kt
package kotlinalgo.string

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*


fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val str = readLine()
    val sb = StringBuilder()

    val set = TreeSet<String>()

    for (i in str.indices) {
        set.add(str.substring(i))
    }

    for (s in set) {
        sb.append(s).append('\n')
    }

    println(sb)

}